src/Step/Deploy/SplitDbConnection.php

```diff
@@ -0,0 +1,251 @@
+     * @param array $connectionsConfig
+     * @return array
+     */
+    private function getMissedSplitTypes(array $splitTypes, array $connectionsConfig): array
+    {
+        return array_values(array_diff_key(
+            array_intersect(self::SPLIT_CONNECTION_MAP, $splitTypes),
+            $this->getSplitConnectionsConfig($connectionsConfig)
+        ));
+    }
+
+    /**
+     * Returns configurations of split connections
+     *
+     * @param array $connectionsConfig
+     * @return array
+     */
+    private function getSplitConnectionsConfig(array $connectionsConfig): array
+    {
+        return array_intersect_key($connectionsConfig, array_flip(DbConfig::SPLIT_CONNECTIONS));
+    }
+
+    /**
+     * Enables split database
+     *
+     * @param array $types
+     * @param array $dbConfig
+     */
+    private function enableSplitConnections(array $types, array $dbConfig)
+    {
+        $splitTypeMap = array_flip(self::SPLIT_CONNECTION_MAP);
+        foreach ($types as $type) {
+            $connectionConfig = $dbConfig[DbConfig::KEY_CONNECTION][$splitTypeMap[$type]];
+            $cmd = $this->buildSplitDbCommand($type, $connectionConfig);
+            $this->logger->debug($this->magentoShell->execute($cmd)->getOutput());
+            $this->logger->info(sprintf(
+                'Quote tables were split to DB %s in %s',
+                $connectionConfig['dbname'],
+                $connectionConfig['host']
+            ));
+        }
+    }
+
+    /**
+     * Returns Magento CLI for split database
+     *
+     * @param string $type
+     * @param array $connectionConfig
+     * @return string
+     */
+    private function buildSplitDbCommand(string $type, array $connectionConfig): string
+    {
+        return sprintf(
+            'setup:db-schema:split-%s --host="%s" --dbname="%s" --username="%s" --password="%s"',
+            $type,
+            $connectionConfig['host'],
+            $connectionConfig['dbname'],
+            $connectionConfig['username'],
+            $connectionConfig['password']
+        );
+    }
+
+    /**
+     * Updates slave connections
+     *
+     * @param array $dbConfig
+     * @throws \Magento\MagentoCloud\Filesystem\FileSystemException
+     */
+    private function updateSlaveConnections(array $dbConfig)
+    {
+        if (!$this->stageConfig->get(DeployInterface::VAR_MYSQL_USE_SLAVE_CONNECTION)) {
+            return;
+        }
+        $mageConfig = $this->configReader->read();
+        $mageConnectionsConfig = $mageConfig[DbConfig::KEY_DB][DbConfig::KEY_CONNECTION];
+        foreach (array_keys($this->getSplitConnectionsConfig($mageConnectionsConfig)) as $mageConnectionName) {
+            if (!isset($dbConfig[DbConfig::KEY_SLAVE_CONNECTION][$mageConnectionName])) {
+                $this->logger->warning(sprintf(
+                    'Slave connection for %s connection not set. '
+                    . 'Relationships not has configuration for this slave connection',
```

Comment:
`not has` -> `do not have the configuration`

Reply:
Done